_playwright-tests/Integration/InstallUploadRepoContent.spec.ts

```diff
@@ -0,0 +1,129 @@
+
+    await test.step('Create upload repository', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.locator('div[id^="pf-modal-part"]').first()).toBeVisible();
+      const nameInput = page.getByPlaceholder('Enter name');
+      await nameInput.click();
+      await nameInput.fill(uploadRepoName);
+      await page.getByLabel('Upload', { exact: true }).check();
+      await page.getByRole('button', { name: 'filter architecture' }).click();
+      await page.getByRole('menuitem', { name: 'x86_64' }).click();
+      const versionFilterButton = page.getByRole('button', { name: 'filter OS version' });
+      await versionFilterButton.click();
+      await page.getByRole('menuitem', { name: 'el9' }).click();
+      await versionFilterButton.click();
+      await Promise.all([
+        page.getByRole('button', { name: 'Save and upload content' }).click(),
+        page.waitForResponse(
+          (resp) =>
+            resp.url().includes('/bulk_create/') && resp.status() >= 200 && resp.status() < 300,
+          { timeout: 600000 },
+        ),
+      ]);
+      const filePath = path.join(__dirname, '../UI/fixtures/bear-4.1-1.noarch.rpm');
+      await retry(page, async (page) => {
+        const fileInput = page.locator('input[type=file]').first();
+        await fileInput.setInputFiles(filePath);
+      });
+      await expect(page.getByText('All uploads completed!')).toBeVisible({ timeout: 60000 });
+      await page.getByRole('button', { name: 'Confirm changes' }).click();
+      const row = await getRowByNameOrUrl(page, uploadRepoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    });
+
+    await test.step('Navigate to templates, and create a template with the upload repository', async () => {
+      const templateNamePrefix = 'integration_test_upload_repo';
+      templateName = `${templateNamePrefix}-${randomName()}`;
+      hostname = `RHSMClientTest-${randomName()}`;
+      regClient = new RHSMClient(hostname);
+
+      await test.step('Set up cleanup for templates and RHSM client', async () => {
+        await cleanup.runAndAdd(() => cleanupTemplates(client, templateNamePrefix));
+        cleanup.add(() => regClient.Destroy('rhc'));
+      });
+
+      await navigateToTemplates(page);
+      await closePopupsIfExist(page);
+      await expect(page.getByRole('button', { name: 'Create template' })).toBeVisible();
+      await page.getByRole('button', { name: 'Create template' }).click();
+      await page.getByRole('button', { name: 'filter architecture' }).click();
+      await page.getByRole('menuitem', { name: 'x86_64' }).click();
+      await page.getByRole('button', { name: 'filter OS version' }).click();
+      await page.getByRole('menuitem', { name: 'el9' }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
+      await page.getByRole('button', { name: 'Next', exact: true }).click();
```

Comment:
This may be flaky - You don't know if the second `Next` will hit the same button as the previous one or the one on the next page. You need to wait in between for something. Some tests are doing this already, check it out